Surface_mesh_shortest_path/include/CGAL/Surface_mesh_shortest_path/Surface_mesh_shortest_path.h

```diff
@@ -1570,9 +1570,9 @@ class Surface_mesh_shortest_path
 
           CGAL_assertion(bool(cgalIntersection));
 
-          Point_2* result = boost::get<Point_2>(&*cgalIntersection);
+          const Point_2* result = boost::get<Point_2>(&*cgalIntersection);
 
-          CGAL_assertion(result && "Error, did not get point intersection on path walk to source");
+          if (!result) result=&current->source_image();
```

Comment:
My one hesitation here would be that I'm typically wary of any code that relies specifically on the return type of a method being a const reference rather than by value (its not terribly risky here, its just a convention I typically avoid). Also, spacing around the `=`